### ISSUE TITLE
Modify `FlowOps.expand` to better match its doc

### ISSUE THREAD

**Issue by aanciaes** [aanciaes/ASD-Project1#4]
I would propose that the current implementation of expand has two gotchas:

It is unclear how the extrapolate argument should be used. From the first sentence of the doc one might imply that the iterator is invoked only when the downstream signals demand, and not for the original element itself. I.e. I'd argue that this:
  
```Java
  implicit val system = ActorSystem()
  implicit val m = ActorMaterializer()

  val resultList = Source(List("1", "2", "3", "4")).expand(_ => Iterator("a")).runWith(Sink.seq)

  println(Await.result(resultList, Duration.Inf))
```
Should print out Vector(1, 2, 3, 4), whereas the actual printout is Vector(a, a, a, a).

This behavior does not appear to provide any functionality that couldn't be replicated with a combination of map and splitX steps (and arguably in a more efficient fashion).

(and, to further flog that dead horse, please see the dictionary definition of "extrapolate" - in fact, in an informal poll among several fellow developers, all were tripped up by the aforementioned issue)

There's no possibility to handle the "degenerate" case of the downstream signaling demand for the pipeline while no elements are present. Serendipitously, there's actually a provision for that in the current (2.5.6) Scaladoc:
@param seed Provides the first state for extrapolation using the first unconsumed element
but it's not reflected in the code.

To summarize, I would propose for FlowOps.expand to have either one of the following definitions:

def expand(extrapolate: Out ⇒ Iterator[Out], seed: Option[Out] = None): Repr[Out]
or

def expand(extrapolate: Out ⇒ Iterator[Out], seed: Out): Repr[Out]
where extrapolate would only be used whenever nothing can be pull ed from upstream on demand.

(note the omission of the U parameter, to remove the possibility of abusing expand as a map with additional bells and whistles)
